lib/Sema/CSSimplify.cpp

```diff
@@ -2057,20 +2057,13 @@ static bool isInPatternMatchingContext(ConstraintLocatorBuilder locator) {
 
   while (!path.empty() && path.back().is<LocatorPathElt::TupleType>())
     path.pop_back();
-
```

Comment:
We should remove this all and maybe `!path.empty()` condition is redundant since `llvm::find_if` would not find anything if the path is empty, so we can remove it as well?

Reply:
also seems fine without this :)